lib/string_validator.rb

```diff
@@ -0,0 +1,24 @@
+require 'pry'
+
+class StringValidator
+  def validate(string)
+    stack = []
+    string.split('').each do |character|
+      if library.values.include?(character)
+        stack << character
+      elsif library[character]
+        stack.pop
+      end
+    end
+
+    if stack == []
```

Comment:
This bit feels a bit tacked on, perhaps we should pull it out to its own method?